doc/tutorial.txt

```diff
@@ -256,6 +256,81 @@ inherit the affinity from the task level):
 	}
 }
 
+*** taskgroups
+
+* taskgroup: String. Can be specified at task level or phase level. Currently
+only SCHED_OTHER tasks are supported. An empty taskgroup value ("") is allowed
+and is interpreted as if there is no taskgroup name/value pair given.
+Pre-existing parts of taskgroups are preserved. In case that parts of
+taskgroups specified in the json file already existed, they are not deleted
+during the cleanup phase.
+A binary without taskgroup support ignores any taskgroup related name/value
+pairs.
+
+The following example sets up a task "thread1" to run in taskgroup "/tg1".
```

Comment:
Here is how I can change this:

```Author: Dietmar Eggemann <dietmar.eggemann@arm.com>
Date:   Fri Oct 19 11:26:24 2018 +0200

    Fix: Move the taskgroup examples from docs/tutorial.txt into files under docs/examples/tutorial
    
    Create two dedicated files example9.json (showing a thread running in a
    taskgroup) and example10.json (showing a thread with multiple phases
    running in different taskgroups) under docs/examples/tutorial and
    remove the examples from docs/tutorial.txt.
    
    Signed-off-by: Dietmar Eggemann <dietmar.eggemann@arm.com>

diff --git a/doc/examples/tutorial/example10.json b/doc/examples/tutorial/example10.json
new file mode 100644
index 000000000000..af07a488782b
--- /dev/null
+++ b/doc/examples/tutorial/example10.json
@@ -0,0 +1,42 @@
+{
+       /*
+        * Use case which creates a thread with 3 phases. In each phase the thread
+        * runs for 2ms and then sleeps for 8ms. In phase0 the thread runs in
+        * taskgroup /tg1/tg11. Phase1 does not specify a taskgroup so the thread
+        * will continue to run in taskgroup /tg1/tg11. In phase2 the thread will
+        * run in taskgroup / (the root taskgroup). The use case runs for 2secs or
+        * until it is stopped with Ctrl+C.
+        */
+       "tasks" : {
+               "thread0" : {
+                       "loop" : -1,
+                       "phases" : {
+                               "phase0" : {
+                                       "run" : 20000,
+                                       "sleep" : 80000,
+                                       "taskgroup" : "/tg1/tg11"
+                               },
+                               "phase1" : {
+                                       "run" : 20000,
+                                       "sleep" : 80000
+                               },
+                               "phase2" : {
+                                       "run" : 20000,
+                                       "sleep" : 80000,
+                                       "taskgroup" : "/"
+                               }
+                       }
+               }
+       },
+       "global" : {
+               "duration" : 2,
+               "calibration" : "CPU0",
+               "default_policy" : "SCHED_OTHER",
+               "pi_enabled" : false,
+               "lock_pages" : false,
+               "logdir" : "./",
+               "log_basename" : "rt-app1",
+               "ftrace" : true,
+               "gnuplot" : true
+       }
+}
diff --git a/doc/examples/tutorial/example9.json b/doc/examples/tutorial/example9.json
new file mode 100644
index 000000000000..020328f8281a
--- /dev/null
+++ b/doc/examples/tutorial/example9.json
@@ -0,0 +1,26 @@
+{
+       /*
+        * Use case which creates a thread that runs 2ms then sleep 8ms
+        * in taskgroup /tg1. The use case runs for 2secs or until it
+        * is stopped with Ctrl+C.
+        */
+       "tasks" : {
+               "thread0" : {
+                       "loop" : -1,
+                       "run" :   20000,
+                       "sleep" : 80000,
+                       "taskgroup" : "/tg1"
+               }
+       },
+       "global" : {
+               "duration" : 2,
+               "calibration" : "CPU0",
+               "default_policy" : "SCHED_OTHER",
+               "pi_enabled" : false,
+               "lock_pages" : false,
+               "logdir" : "./",
+               "log_basename" : "rt-app1",
+               "ftrace" : true,
+               "gnuplot" : true
+       }
+}
diff --git a/doc/tutorial.txt b/doc/tutorial.txt
index a3f1f76f9648..7c64787bcd82 100644
--- a/doc/tutorial.txt
+++ b/doc/tutorial.txt
@@ -267,70 +267,6 @@ during the cleanup phase.
 A binary without taskgroup support ignores any taskgroup related name/value
 pairs.
 
-The following example sets up a task "thread1" to run in taskgroup "/tg1".
-
-"tasks" : {
-        "thread1" : {
-                "taskgroup" : "/tg1",
-                "run" : 10,
-                "sleep" : 10
-        }
-}
-
-The following example sets up a task "thread2" with three phases. "thread2"
-will run in taskgroup "/tg1/tg11" in phase1. Phase2 does not specify a
-taskgroup so "thread2" will continue to run in "/tg1/tg11". In phase3 "thread2"
-will run in taskgroup "/" (root).
-
-"tasks" : {
-        "thread2" : {
-                "phases" : {
-                        "phase1" : {
-                                "taskgroup" : "/tg1/tg11",
-                                "run" : 10,
-                                "sleep" : 10
-                        },
-                        "phase2" : {
-                                "run" : 10,
-                                "sleep" : 10
-                        },
-                        "phase3" : {
-                                "taskgroup" : "/",
-                                "run" : 10,
-                                "sleep" : 10
-                        }
-                }
-        }
-}
-
-Specifying taskgroups on task and phase level for a task should be avoided.
-Otherwise, it is possible that a phase of a task runs in different taskgroups.
-For example, phase1 of task "thread3" would run in "/tg1" in the first loop and
-in "/" in the second loop.
-
-"tasks" : {
-        "thread3" : {
-                "loop": 2,
-                "taskgroup" : "/tg1",
-                "phases" : {
-                        "phase1" : {
-                                "run" : 10,
-                                "sleep" : 10
-                        },
-                        "phase2" : {
-                                "taskgroup" : "/tg1/tg11",
-                                "run" : 10,
-                                "sleep" : 10
-                        },
-                        "phase3" : {
-                                "taskgroup" : "/",
-                                "run" : 10,
-                                "sleep" : 10
-                        }
-                }
-        }
-}
-
 *** events ***
 
 events are simple action that will be performed by the thread or on the
```

Reply:
Looks better to me.

Reply:
I put these 2 changes in the upcoming version.